scripts-python/create_test

```diff
@@ -135,128 +140,91 @@ formatter_class=argparse.ArgumentDefaultsHelpFormatter
     if (args.no_build):
         args.no_run = True
 
-    use_batch = cime_util.does_machine_have_batch()
```

Comment:
I find it very confusing to mix the argument parsing with other code, I want to know for example whether the user has input a machine or there is a machine in test request before I select a default compiler